### ISSUE TITLE
Cop idea: mutually exclusive FactoryBot traits

### ISSUE THREAD

**Issue by pirj** [rubocop/rubocop-factory_bot#14]
```ruby
FactoryBot.define do
  factory :user
  trait :active do
    status { 'active' }
  do
  trait :inactive do
    status 'inactive'
  do
end
```
This allows for an undefined behaviour when being called as `create(:user, :active, :inactive)`.
We should recommend making them sub-factories, not traits.

**Comment by ydah** [comment]
Indeed, that case can't be specified. It seems prone to false positives. Shouldn't we close this? This should be addressed in the team guidelines or documentation, not via cop.

**Comment by pirj** [comment]
> team guidelines or documentation

Agree. This is hard to formalize.
